stackit/internal/services/argus/scrapeconfig/resource.go

```diff
@@ -582,47 +709,52 @@ func toCreatePayload(ctx context.Context, model *Model) (*argus.CreateScrapeConf
 		SampleLimit: utils.Ptr(float64(model.SampleLimit.ValueInt64())),
 		Scheme:      conversion.StringValueToPointer(model.Scheme),
 	}
-	setDefaultsCreateScrapeConfig(&sc, model)
+	setDefaultsCreateScrapeConfig(&sc, model, saml2Model)
 
-	if model.SAML2 != nil && !model.SAML2.EnableURLParameters.ValueBool() {
+	if !saml2Model.EnableURLParameters.IsNull() && !saml2Model.EnableURLParameters.IsUnknown() {
 		m := make(map[string]interface{})
 		if sc.Params != nil {
 			m = *sc.Params
 		}
-		m["saml2"] = []string{"disabled"}
+		if saml2Model.EnableURLParameters.ValueBool() {
+			m["saml2"] = []string{"enabled"}
```

Comment:
isn't this already set on `defaultsCreateScrapeConfig`?

Reply:
then it could be added to the if clause like on the original code

Reply:
Yes, but before we were relying on the default to be true, this now works for both possible default cases